### PR TITLE
ActionBar docs

### LINKED ISSUE #2771: ActionBar documentation
# 🙋 Documentation Request

We need to write docs for ActionBar 

## 🧢 Your Company/Team

RSP

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes #2771 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 🧢 Your Project:

RSP

### CODE REVIEW

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (LFDanLu) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:23]: Is Collections the right category (felt most closely related to those components)? Maybe Overlays?

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (LFDanLu) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:42]: This example takes a lot of vertical space unfortunately. What do people think about me reducing this and the dynamic example to 
```
    <ActionBarContainer>
      <>
      <ActionBar
        isEmphasized
        selectedItemCount={selectedKeys === 'all' ? selectedKeys : selectedKeys.size}
        onClearSelection={() => {
          setSelectedKeys(new Set());
        }}>
        <Item key="edit">
          Edit
        </Item>
        <Item key="copy">
          Copy
        </Item>
        <Item key="delete">
          Delete
        </Item>
      </ActionBar>
    </ActionBarContainer>
```
so that it only shows the ActionBarContainer? Then the example in the `Events` section I can include the TableView and it won't be the same example basically used 4 times...

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (LFDanLu) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:178]: Perhaps extraneous section?

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (LFDanLu) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:216]: A bit strange because it isn't a "realistic" example (no accompanying element other than the ActionBars) but just trying to not make it take up so much vertical space (aka CardView and TableView are very tall code wise and physically)

File: packages/@react-spectrum/actionbar/docs/ActionBarAnatomy.svg
Comment (LFDanLu) [inline packages/@react-spectrum/actionbar/docs/ActionBarAnatomy.svg:1]: the anatomy stuff is a bit small on mobile, trying to fix...

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (LFDanLu) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx]: Sounds good, I'll move it to Overlay

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (LFDanLu) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx]: Updated the examples, people can compare and contrast the two builds

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (snowystinger) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx]: hmmm but this isn't an overlay, at least, not in the classic sense. It's rendered inline and space is made for it using flex.
I think I might consider a different category, a new one, something like a companion, container, wrapper, extra functionality, bulk actions, collection actions, actions, .... naming is hard, haha

Comment (LFDanLu) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx]: Thats a good point, I had figured that it looked like an Overlay although not truly being one and thus people would look for it in that category there. I like "Collection Actions" personally, open to other options

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (LFDanLu) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx:64]: Perhaps just `An ActionBar is a companion component that consists of an ActionGroup, a clear button, and a count of selected items.`? Or maybe just `An ActionBar consists of an ActionGroup, a clear button, and a count of selected items.`

Comment (snowystinger) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx:64]: I think if you say companion component, you need to expand on that and add something to the effect of, "it is intended to be used with another component"
Maybe that should be added in either case

File: packages/@react-spectrum/actionbar/src/ActionBar.tsx
Comment (snowystinger) [inline packages/@react-spectrum/actionbar/src/ActionBar.tsx:59]: should we pull this out into a separate PR?

**Review by snowystinger — DISMISSED** [review]

**Review by majornista — DISMISSED** [review]

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (ktabors) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx]: I feel like this is similar to an ActionMenu on a card, which would be a collection item action. I'm trying to figure out what else other things like this would be.

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (ktabors) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:23]: I just noticed that `ActionMenu` docs were placed in "Collections". I was looking at the SideNav and I wanted an "Actions" section for `ActionGroup`, `ActionBar`, and `ActionMenu`. Maybe `ActionButton` too.

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (ktabors) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:97]: Should this sentence be more explicated that the TableView height is overridden by the ActionBarContainer height?

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (ktabors) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:160]: Should we have an example that uses icons and aria-label? I don't see that elsewhere on this page. The example in ActionGroup does this. I'm wondering if we need to be explicate or repeat ourselves?

File: packages/@react-types/actionbar/src/index.d.ts
Comment (ktabors) [inline packages/@react-types/actionbar/src/index.d.ts:16]: I didn't like the definition of `items` in the props table, but it is generic coming from the CollecitonBase. The `items` are for the ActionGroup so the definition seems wrong, but I don't think we can fix it.

**Review by ktabors — COMMENTED** [review]
Very cool!

File: packages/@react-spectrum/actionbar/src/ActionBar.tsx
Comment (LFDanLu) [reply packages/@react-spectrum/actionbar/src/ActionBar.tsx:133]: Perhaps "needs to perform actions on one or more items at the same time"

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (LFDanLu) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx:97]: to be honest, I'm not sure if that is the desired behavior now that I look at it again. Perhaps the TableView should preserve its own height?

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (LFDanLu) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx:160]: I figure users would look at the ActionGroup docs for that, don't wanna repeat too much if at all possible IMO.

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (ktabors) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx:23]: Talked in Slack and we arrived on `Toolbars` as the section for this component.

File: packages/@react-spectrum/actionbar/src/ActionBar.tsx
Comment (snowystinger) [reply packages/@react-spectrum/actionbar/src/ActionBar.tsx:59]: revisiting this, do we need to update the after_version? or does this need to be pulled out?

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (snowystinger) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:168]: I think we need to put a height on the table? the usual use case is that the table shrinks upwards to make room for the actionbar instead of the page shifting down to make room for it

**Review by snowystinger — COMMENTED** [review]
given current contrast ratio issues with emphasized, should we change all the default examples in this documentation to be the non-emphasized version?

**Review by snowystinger — COMMENTED** [review]
we've on 3.0.0-alpha.14 now, so this will be live immediately right now
after_version: 3.0.0-alpha.4

File: packages/@react-spectrum/actionbar/src/ActionBar.tsx
Comment (reidbarber) [reply packages/@react-spectrum/actionbar/src/ActionBar.tsx:59]: Updated the after_version. I can pull this out if we need to though. Not sure what the priority of this  component is.

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (ktabors) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:24]: collections?

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (ktabors) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:65]: `... of an ActionGroup of actions, a clear...`?
Should order of ActionGroup, button, and count mirror the anatomy diagram order?

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (ktabors) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:93]: This example isn't rendering. And most of the examples after this too. (Reid pointed out this is due to removing the code change.)

Comment (reidbarber) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx:94]: should be fixed now

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (reidbarber) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx]: added

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (reidbarber) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx:66]: I think the current order is fine since the ActionGroup is the most important part of the component.

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (LFDanLu) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:64]: Small nit, we'll need an aria-label and role="img" on this anatomy diagram, can look at the dnd.mdx docs for an example.

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (LFDanLu) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:105]: Maybe this example should have ` onClearSelection` hooked up? While that functionality is demonstrated in the events section below, it feels like users may just see this example first and be confused as to why the buttons don't seem to do anything to selection.

Happy to go either way, bit torn on whether or not it is necessary

**Review by LFDanLu — COMMENTED** [review]
Definitely can/should be followup, but it would be nice to start using the highlight syntax that gets used in the [ListView DnD code examples](https://react-spectrum.adobe.com/react-spectrum/ListView.html#drag-and-drop) and other places.

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (ktabors) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx:105]: Including `onClearSelection` makes sense. More so than hooking up the buttons which I was concerned about, since they also do nothing.

Comment (reidbarber) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx:105]: Updated

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (reidbarber) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx]: Added

**Review by ktabors — DISMISSED** [review]

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (LFDanLu) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:146]: IMO this highlight should be on the height applied to the ActionBarContainer since that is what the layout section is mentioning specifically

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (LFDanLu) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:227]: Maybe highlight these two event handlers specifically to make it more specific to the section this example is under

**Review by LFDanLu — COMMENTED** [review]
Last two nitpicks, I promise!

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (reidbarber) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx:146]: Highlighting JSX only works when you wrap an entire element unfortunately.

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (reidbarber) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx:227]: Same as above

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (reidbarber) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx:146]: I guess you can do this:

```jsx
    <ActionBarContainer
      /*- begin highlight -*/
      height="size-5000"
      /*- end highlight -*/>
```

but the indention isn't quite right:

Screen Shot 2022-12-01 at 1 26 52 PM

Looking into it.

Comment (reidbarber) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx]: Updated to this since it's better than before IMO

**Review by LFDanLu — COMMENTED** [review]
LGTM, thanks for updating @reidbarber. Provisional approval since I can't approve my own PR, feel free to approve in my stead haha

**Review by reidbarber — APPROVED** [review]

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (ktabors) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx]: Wouldn't we want to highlight the TableView height, to draw attention to it being small then the ActionBar height? I know we are only talking about the ActionBar height here, but in this example it matters that the two values are different.

Comment (reidbarber) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx]: Updated to include this.

**Review by reidbarber — DISMISSED** [review]

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (devongovett) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:61]: I thought we were going to change all the examples so there was always a TableView or ListView with the ActionBar. A standalone action bar would never exist in a real app.

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (reidbarber) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx]: I agree, I'll remove this

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (reidbarber) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx]: Updating all to include list or table

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (reidbarber) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx:89]: Updated so that all in include a list or table.

**Comment by reidbarber** [comment]
reidbot: https://reactspectrum.blob.core.windows.net/reactspectrum/53aaa1cecc276d0a241020ec938c27eda604fdd8/docs/react-spectrum/ActionBar.html

File: packages/@react-types/actionbar/src/index.d.ts
Comment (ktabors) [inline packages/@react-types/actionbar/src/index.d.ts:30]:
```suggestion
  /** The contents of the ActionBarContainer. Should include a ActionBar and the renderable content it is associated with. */
```
The lint rule seems to only complain about no space at the start and not at the end.

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (LFDanLu) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:116]: Nit: Probably should change the aria-label here to distinguish it from the example above
```suggestion
      <ListView items={listItems} aria-label="ListView with action bar, dynamic collection" selectionMode="multiple" selectedKeys={selectedKeys} onSelectionChange={setSelectedKeys}>
```

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (LFDanLu) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:181]: Nit
```suggestion
      <TableView aria-label="Table with action bar and controlled selection" selectionMode="multiple" selectedKeys={selectedKeys} onSelectionChange={setSelectedKeys} height="size-3000">
```

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (LFDanLu) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:275]: Nit:
The example should have separate aria-labels to distinguish them from one another. One should mention it is a emphasized actionbar

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (snowystinger) [inline packages/@react-spectrum/actionbar/docs/ActionBar.mdx:32]: just a reminder we'll need to update this after we do release, would be good to open the PR updating this after this one is merged so we don't forget

**Review by snowystinger — APPROVED** [review]
Will re-approve after current comments are addressed

File: packages/@react-spectrum/actionbar/docs/ActionBar.mdx
Comment (reidbarber) [reply packages/@react-spectrum/actionbar/docs/ActionBar.mdx:253]: Good idea, added this.

**Review by ktabors — APPROVED** [review]

**Comment by LFDanLu** [comment]
LGTM, @reidbarber if you'd like to approve in my place + merge

**Review by reidbarber — APPROVED** [review]

**Pull request merged**
